scripts/sonar/sonar-controller.js

```diff
@@ -41,23 +50,51 @@ angular.module('app.sonar').controller('SonarCtrl', ['$scope', '$log', '$q', 'ap
                 var totalLines = sum(timePeriod.metrics, 'lines');
 
                 var weightedMetrics = timePeriod.metrics.map(function(project) {
-                    var factor = project.lines / totalLines;
 
                     return {
-                        methodComplexity: factor * project.methodComplexity,
-                        fileComplexity: factor * project.fileComplexity,
-                        coverage: factor * project.coverage,
-                        compliance: factor * project.compliance,
+                        numberOfMethodsInProjectInPeriod: project.totalMethods,
+                        numberOfFilesInProjectInPeriod: project.totalFiles,
+                        complexityInProjectInPeriod: project.totalComplexity,
+                        testableLinesInProjectInPeriod: project.testableLines,
+                        untestedLinesInProjectInPeriod: project.untestedLines,
+
+                        weightedBlockingIssuesInProjectInPeriod: weighting.blocker * project.blockerIssues,
+                        weightedCriticalIssuesInProjectInPeriod: weighting.critical * project.criticalIssues,
+                        weightedMajorIssuesInProjectInPeriod: weighting.major * project.majorIssues,
+                        weightedMinorIssuesInProjectInPeriod: weighting.minor * project.minorIssues,
+                        weightedInfoIssuesInProjectInPeriod: weighting.info * project.infoIssues
+
                     };
                 });
-
+                
+                // Complexity calculations
```

Comment:
The claculations would probably be nice in 2 functions, one for complexity and one for compliance, just to cut down the length of this function a bit.